Lib/test/test_exceptions.py

```diff
@@ -1695,6 +1695,19 @@ def __getattr__(self, attr):
 
             self.assertIn("blech", err.getvalue())
 
+    def test_attribute_error_with_failing_dict(self):
+        class T:
+            bluch = 1
+            def __dir__(self):
+                return self.blech
```

Comment:
I would prefer to explicitly raise an exception. It's hard to guess the intent here.

Reply:
Done